lib/common/AsyncTimeout.ts

```diff
@@ -0,0 +1,46 @@
+/**
```

Comment:
Few suggestions:
1. The class name to something like: ````AsyncExecutor````?
1. The function name to something like: ````ExecuteWithTimeout````?
1. The signature: ````public static async ExecuteWithTimeout<T> (asyncCall: Promise<T>, timeoutInMilliseconds: number): Promise<T>````
1. Upon reaching timeout, the function just throws ````AsyncTimeoutError````. Otherwise, T is returned. Or I guess we can return the ````AsyncExecuteWithTimeoutResult```` if we want to stick with your current model.
1. For any other errors, an ````IpfsError```` is thrown?
1. All the related classes for this functionality is under ````async```` folder.